### ISSUE TITLE
cloud-functions-start (step 8): An unexpected error has occurred.

### ISSUE THREAD

**Issue by oneezy** [firebase/friendlychat-web#222]
When trying to [Welcome New Users](https://codelabs.developers.google.com/codelabs/firebase-cloud-functions/#7) in Step 8 of the **cloud-functions-start** code lab, I'm getting an error after I run the following in the terminal:

**Command:**
```
firebase deploy --only functions
```

**Output:**
```
=== Deploying to 'www-fire'...

i  deploying functions
i  functions: ensuring necessary APIs are enabled...
i  runtimeconfig: ensuring necessary APIs are enabled...
✔  functions: all necessary APIs are enabled
✔  runtimeconfig: all necessary APIs are enabled
i  functions: preparing functions directory for uploading...

Error: An unexpected error has occurred.
```
---
Any idea what the problem could be?

**Comment by kroikie** [comment]
Hi @oneezy your deploy command looks fine, is this still a problem? I just want to confirm that it is not an issue on the server side that has been resolved and is something I need to look into further.

**Comment by mismith** [comment]
I'm also experiencing this :(

**Comment by nicolasgarnier** [comment]
This issue was moved to firebase/friendlychat-android#1

**Comment by kevando** [comment]
try running with --debug

https://github.com/firebase/firebase-tools/issues/356

I came across the same issue and found it was because I had Node 8.0.0

I used nvm to upgrade to 8.1.0 and opened a new terminal window and it appears to have worked..


Might break other shit that depends on node 8.0.0 but we shall see..

**Comment by nicolasgarnier** [comment]
:+1: @kevando There was an issue with backward compatibility in Node 8.0.0 in particular which got fixed in 8.1.0.

Currently it's best to use 6.x.x latest when using Google Cloud Functions as this is what is supported and runs on prod. I with the latest Firebase CLI you should now get a warning about this when trying to run `firebase serve`

**Comment by DiabloRodriguez** [comment]
Oh ok, f*cking "sudo" command. For all Linux users, just add them at the begin => "sudo firebase serve"
